hail/src/test/scala/is/hail/io/fs/FSSuite.scala

```diff
@@ -42,45 +44,56 @@ trait FSSuite extends TestNGSuite {
 
   def pathsRelResourcesRoot(statuses: Array[FileListEntry]): Set[String] = pathsRelRoot(fsResourcesRoot, statuses)
 
-  @Test def testExists(): Unit = {
-    assert(fs.exists(r("/a")))
-
-    assert(fs.exists(r("/zzz")))
-    assert(!fs.exists(r("/z"))) // prefix
-
+  @Test def testExistsOnDirectory(): Unit = {
     assert(fs.exists(r("/dir")))
     assert(fs.exists(r("/dir/")))
 
     assert(!fs.exists(r("/does_not_exist")))
     assert(!fs.exists(r("/does_not_exist_dir/")))
   }
 
+  @Test def testExistsOnFile(): Unit = {
+    assert(fs.exists(r("/a")))
+
+    assert(fs.exists(r("/zzz")))
+    assert(!fs.exists(r("/z"))) // prefix
+  }
+
+  @Test def testFileStatusOnFile(): Unit = {
+    // file
+    val f = r("/a")
+    val s = fs.fileStatus(f)
+    assert(s.getPath == f)
+    assert(s.getLen == 12)
+  }
+
   @Test def testFileListEntryOnFile(): Unit = {
     // file
     val f = r("/a")
     val s = fs.fileListEntry(f)
     assert(s.getPath == f)
-    assert(s.isFile)
-    assert(!s.isDirectory)
     assert(s.getLen == 12)
   }
 
+  @Test def testFileStatusOnDirIsFailure(): Unit = {
+    val f = r("/dir")
+    TestUtils.interceptException[FileNotFoundException](r("/dir"))(
+      fs.fileStatus(r("/dir"))
```

Comment:
I thought you could create a fileStatus object for a directory. It just doesn't have the directory information as a FLE would have.

Reply:
You can only create a fileStatus for a file/object not a directory/prefix. In particular, no single Class B Operation can reliably determine if a path is a directory. For example, if a GET for the object `gs://bucket/dir/a` returns 200, I know that `gs://bucket/dir` is a directory but if it returns 404, I don't know if that directory does or does not exist.